src/rule_checker/rule_checker.ts

```diff
@@ -150,6 +152,11 @@ export class RuleChecker implements IRuleChecker {
                     }
                 }
             }
+
+            if (existingPID === childPID) {
```

Comment:
any reason why we do this check after the loop? We existingPID and childPID don;'t change so this check can be done earlier and potentially save some cycles

Reply:
No reason, I can move that up